kyaml/kio/byteio_reader.go

```diff
@@ -110,20 +119,26 @@ func (r *ByteReader) Read() ([]*yaml.RNode, error) {
 			continue
 		}
 		if err != nil {
-			return nil, err
+			return nil, errors.Wrap(err)
 		}
 		if yaml.IsMissingOrNull(node) {
 			// empty value
 			continue
 		}
 
 		// ok if no metadata -- assume not an InputList
-		meta, _ := node.GetMeta()
+		meta, err := node.GetMeta()
+		if err != yaml.ErrMissingMetadata && err != nil {
+			return nil, errors.WrapPrefixf(err, "[%d]", i)
+		}
 
 		// the elements are wrapped in an InputList, unwrap them
 		// don't check apiVersion, we haven't standardized on the domain
-		if (meta.Kind == ResourceListKind || meta.Kind == "List") &&
+		if !r.DisableUnwrapping &&
```

Comment:
This is an unrelated bug fix.  LMK if you want me to pull it into a separate commit.